libnmstate/nm/device.py

```diff
@@ -378,6 +375,10 @@ def _delete_device_callback(src_object, result, user_data):
         if mainloop.is_action_canceled(e):
             logging.debug('Device deletion aborted on %s: error=%s',
                           nmdev.get_iface(), e)
+        elif 'No such interface' in str(e):
```

Comment:
How about:
```python
exept GLib.GError as e:
    if e.domain == 'g-dbus-error-quark' and e.code == 19:
...
```

Reply:
It makes sense, but we have a few problems popping up:
- pylint does not like the GLib.GError attributes (domain, code).
- It is unclear to me why the code is errno.ENODEV related. I could not find anything that specify that the error code has anything to do with the codes from errno.

The first point can be handled (hopefully) as we did in `is_action_canceled()`.

Reply:
I think the correct way to test for an unregistered device is to check for the unknown-method error code of gdbus:

``
if e.matches(Gio.DBusError.quark(), Gio.DBusError.UNKNOWN_METHOD):
``

Reply:
`errno.ENODEV` is by my guess. Clearly @bengal is right on the `Gio.DBusError.UNKNOWN_METHOD`.

Reply:
Done